blueflood-http/src/main/java/com/rackspacecloud/blueflood/outputs/handlers/HttpMultiRollupsQueryHandler.java

```diff
@@ -111,14 +111,15 @@ public void handle(ChannelHandlerContext ctx, HttpRequest request) {
             return;
         }
 
-        HTTPRequestWithDecodedQueryParams requestWithParams = (HTTPRequestWithDecodedQueryParams) request;
+        List<String> locatorStrings = new ArrayList<String>();
+        for (Locator locator : locators)
+            locatorStrings.add(locator.toString());
 
+        HTTPRequestWithDecodedQueryParams requestWithParams = (HTTPRequestWithDecodedQueryParams) request;
         final Timer.Context httpBatchMetricsFetchTimerContext = httpBatchMetricsFetchTimer.time();
         try {
             RollupsQueryParams params = PlotRequestParser.parseParams(requestWithParams.getQueryParams());
-            BatchMetricsQuery query = new BatchMetricsQuery(locators, params.getRange(), params.getGranularity());
-            Map<Locator, MetricData> results = new BatchMetricsQueryHandler(executor, AstyanaxReader.getInstance())
-                                                        .execute(query, queryTimeout);
+            Map<Locator, MetricData> results = getRollupByGranularity(tenantId, locatorStrings, params.getRange().getStart(), params.getRange().getStop(), params.getGranularity());
```

Comment:
In BatchMetricsQueryHandler, we track a few metrics such as 

```
"Batched Metrics Query Duration");
    private static final Meter exceededQueryTimeout = Metrics.meter(Instrumentation.class,
            "Batched Metrics Query Duration Exceeded Timeout");
    private static final Histogram queriesSizeHist = Metrics.histogram(Instrumentation.class,
            "Total queries"
```

Do we have dashboards for these? I was wondering if getting all the processing out in the netty thread will not block it for a long time?